### ISSUE TITLE
Screen Reader issues with Checklists

### ISSUE THREAD

**Issue by azollafox** [Automattic/Simplenote-United#142]
### Expected
A screenreader should be able to tell a user when they have selected a checklist

### Observed

>When I select checklist, only something like symbol 0 appears next to the item, this is how the screen reader interprets it. I don't know if it's a checkbox, and I don't know if when I tap it, it gets checked or nothing happens.

### Where did you see the bug
 * Desktop: Windows 10, version 20H2, screenreader NVDA 2022.2.2,
Simplenote version 2.21.0
* Mobile: Android 12, TalkBack version 13, Simplenote version 2.24.1 (150)

user reported 5495060-zen
